### PR TITLE
Fix AWS system test example_dynamodb_to_s3

### PR DESCRIPTION
**Pull request by vincbeck**
The system test `example_dynamodb_to_s3` is currently failing for multiple reasons:
- The point in time recovery must be enabled in the table
- Once enabled, you need to get a valid export time. You can get such information by calling `describe_continuous_backups` API.

See related: #31307

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by vincbeck** [comment]
cc @utkarsharma2

File: tests/system/providers/amazon/aws/example_dynamodb_to_s3.py [hunk 1]
```diff
@@ -56,9 +56,24 @@ def set_up_table(table_name: str):
...
     table.put_item(Item={"ID": "123", "Value": "Testing"})
 
 
+@task
+def get_export_time(table_name: str):
+    r = boto3.client("dynamodb").describe_continuous_backups(
+        TableName=table_name,
+    )
+
+    return r["ContinuousBackupsDescription"]["PointInTimeRecoveryDescription"]["EarliestRestorableDateTime"]
```
Comment (ferruzzi) [inline tests/system/providers/amazon/aws/example_dynamodb_to_s3.py:74]: Oh, that's nicer than my created time idea.  Nice find.

File: airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py [hunk 2]
```diff
@@ -163,7 +168,7 @@ def _export_table_to_point_in_time(self):
 
     def _export_entire_data(self):
         """Export all data from the table."""
-        table = self.hook.get_conn().Table(self.dynamodb_table_name)
+        table = self.hook.conn.Table(self.dynamodb_table_name)
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py:171]: Thanks.

File: airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py [hunk 3]
```diff
@@ -97,9 +97,14 @@ class DynamoDBToS3Operator(AwsToAwsBaseOperator):
...
         *AwsToAwsBaseOperator.template_fields,
+        "dynamodb_table_name",
         "s3_bucket_name",
+        "file_size",
+        "dynamodb_scan_kwargs",
         "s3_key_prefix",
-        "dynamodb_table_name",
+        "process_func",
+        "export_time",
+        "export_format",
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/transfers/dynamodb_to_s3.py:107]: Thanks.

**Review by ferruzzi — APPROVED** [review]

**Comment by ferruzzi** [comment]
One of your unit tests are failing, bit otherwise LGTM.  Some nice nitpick catches in there as well, thanks for those.

File: tests/system/providers/amazon/aws/example_dynamodb_to_s3.py (hunk 1 above)
Comment (utkarsharma2) [reply tests/system/providers/amazon/aws/example_dynamodb_to_s3.py:74]: This looks good! thanks for taking care of this.

**Review by utkarsharma2 — APPROVED** [review]

**Comment by utkarsharma2** [comment]
@vincbeck Thanks for handling the timezone scenario as well. LGTM :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
(Approved - pending the unit-test fix :)

**Comment by vincbeck** [comment]
All green :)

**Comment by vincbeck** [comment]
:D

Screenshot 2023-05-17 at 4 56 57 PM

**Comment by potiuk** [comment]
🎉 🎉 🎉 🎉 🎉

**Pull request merged**